packages/react-sdk/src/index.tsx

```diff
@@ -9,19 +10,58 @@ import canonicalJSON from "canonical-json";
 
 import type {
   FeatureFlagsOptions,
-  Flags,
   Options as BucketSDKOptions,
 } from "@bucketco/tracking-sdk";
 import BucketSingleton from "@bucketco/tracking-sdk";
+import {
+  Feedback,
+  RequestFeedbackOptions,
+} from "@bucketco/tracking-sdk/dist/types/src/types";
 
 export type BucketInstance = typeof BucketSingleton;
 
+type UserContext = {
+  id: string | number;
+  [key: string]: any;
+};
+
+type CompanyContext = {
+  id: string | number;
+  [key: string]: any;
+};
+
+type OtherContext = Record<string, any>;
+
+type FlagContext = {
+  user?: UserContext;
+  company?: CompanyContext;
+  otherContext?: OtherContext;
+};
+
+export interface Flags {}
+
+type BucketFlags = keyof (keyof Flags extends never
+  ? Record<string, boolean>
+  : Flags);
+
+export type FlagsResult = { [k in BucketFlags]?: boolean };
+
 type BucketContext = {
   bucket: BucketInstance;
   flags: {
-    flags: Flags;
+    flags: FlagsResult;
     isLoading: boolean;
   };
+  updateUser: (user?: UserContext) => void;
+  updateCompany: (company: CompanyContext) => void;
+  updateOtherContext: (otherContext: OtherContext) => void;
+
+  sendFeedback: (opts: Omit<Feedback, "userId" | "companyId">) => void;
+  requestFeedback: (
+    opts: Omit<RequestFeedbackOptions, "userId" | "companyId">,
+  ) => void;
+
+  track: (eventName: string, attributes?: Record<string, any>) => void;
 };
 
 const Context = createContext<BucketContext>({
```

Comment:
I would consider calling this `BucketContext` instead and the type `BucketContextState` or similar.

Reply:
since they are not exported and everything in this file is "bucket" i went for `ProviderContext` and `ProviderContextType`. I assume you commented because `context` could be conflated with flag context etc. ?

Reply:
I had a chat with @roncohen and we both agreed there were benefits and detriments to grouping it under one hook. Ultimately we decided to stick with the current approach and to test it in BETA first to get a feel for it.